### PR TITLE
fix: Cpp openrpc updates

### PR DESCRIPTION
**Pull request by ksentak**

### CODE REVIEW

**Review by kevinshahfws — APPROVED** [review]

File: languages/c/templates/json-types/property-assign.cpp [hunk 1]
```diff
@@ -1,2 +1,3 @@
         Add(_T("${property}"), &${Property});
         ${Property} = other.${Property}
+        members[${property.raw}] = ${Property}
```
Comment (kschrief) [inline languages/c/templates/json-types/property-assign.cpp:3]: Leftover from my efforts that can be removed.

File: languages/cpp/templates/methods/property.cpp [hunk 2]
```diff
@@ -4,7 +4,7 @@
         const string method = _T("${info.title.lowercase}.${method.name}");
         ${if.params}JsonObject jsonParameters;${end.if.params}
         ${if.params}${method.params.serialization}${end.if.params}
-        ${method.result.json} jsonResult;
+        ${method.result.json.type} jsonResult;
```
Comment (kschrief) [inline languages/cpp/templates/methods/property.cpp:7]: Leftover from my efforts that can (probably) be removed.

File: src/macrofier/types.mjs [hunk 3]
```diff
@@ -348,8 +348,12 @@ const insertObjectMacros = (content, schema, module, title, property, options) =
         // don't push properties w/ unsupported types
         if (type) {
           const description = getSchemaDescription(prop, module)
+          const additionalType = typeof prop.type == 'string' ? getPrimitiveType(prop.type, 'additional-types') : null
```
Comment (kschrief) [inline src/macrofier/types.mjs:351]: Leftover from my efforts that can be removed.

File: src/macrofier/types.mjs [hunk 4]
```diff
@@ -348,8 +348,12 @@ const insertObjectMacros = (content, schema, module, title, property, options) =
...
           let replacedTemplate  = template
           .replace(/(^\s+)/g, '$1'.repeat(options2.level))
+          .replace(/\$\{additional\.type\}/g, additionalType) //C++ - Gives the object type (Ex: Boolean()). Non-AnyOf only
```
Comment (kschrief) [inline src/macrofier/types.mjs:354]: Leftover from my efforts that can be removed.

**Review by kschrief — CHANGES_REQUESTED** [review]

**Review by kschrief — APPROVED** [review]

**Pull request merged**